### PR TITLE
Refactored Delete feature

### PR DESCRIPTION
**Pull request by Pegasusx10**

### CODE REVIEW

File: students.js [hunk 1]
```diff
@@ -34,35 +35,65 @@ router.post('/', async (req, res) => {
...
+  res.json(identity)
+});
+
+// try{
+//   const id = req.params.id;
+//   const firstName = req.body.firstName;
+//   const lastName = req.body.lastName;
+//   const grade = req.body.grade;
+//   const division = req.body.division;
+//   const options = {new: true};
```
Comment (nandini-vel) [inline students.js:69]: @Pegasusx10 commented code should not be merged to develop

File: students.js [hunk 2]
```diff
@@ -34,35 +35,65 @@ router.post('/', async (req, res) => {
...
+    })
+  })
+  .catch(err=>{
+    res.status(500).json({
+      error:err
+    })
+  })
 })
 
 async function getstudent(req, res, next) {
```
Comment (nandini-vel) [inline students.js:93]: getStudent

File: server.js [hunk 3]
```diff
@@ -3,6 +3,7 @@ require('dotenv').config()
 const express = require('express')
 const app = express()
 const mongoose = require('mongoose')
+const student = require('./api/student')
 
 mongoose.connect('mongodb+srv://Pegasusx10:pegasus123@cluster0.749smlf.mongodb.net/student-database?retryWrites=true&w=majority', { useUnifiedTopology: true, useNewUrlParser: true })
```
Comment (nandini-vel) [inline server.js:8]: urls should not be hardcoded. They should be picked from an environment config file. database connection should be done separately and that connection should be reused in all files

**Review by nandini-vel — CHANGES_REQUESTED** [review]

**Review by adarshajit — CHANGES_REQUESTED** [review]

**Review by adarshajit — CHANGES_REQUESTED** [review]

**Review by adarshajit — CHANGES_REQUESTED** [review]

File: server.js [hunk 4]
```diff
@@ -3,15 +3,17 @@ require('dotenv').config()
 const express = require('express')
 const app = express()
 const mongoose = require('mongoose')
-
-mongoose.connect('mongodb+srv://Pegasusx10:pegasus123@cluster0.749smlf.mongodb.net/student-database?retryWrites=true&w=majority', { useUnifiedTopology: true, useNewUrlParser: true })
+const studentRouter = require('./routes/students')
+const connection = require('./config/dbconnection')
```
Comment (nandini-vel) [inline server.js:7]: dbConnection

File: students.js [hunk 5]
```diff
@@ -2,26 +2,45 @@ const express = require('express')
 const router = express.Router()
 const student = require('../api/student')
 
-// Getting all 
+// Getting all Pagination and Query
 router.get('/', async (req, res) => {
   try {
-    const students = await student.find()
-    res.json(students)
+const pageSize = parseInt(req.query.pageSize) || 0
```
Comment (nandini-vel) [inline students.js:8]: fix alignment

**Review by adarshajit — COMMENTED** [review]
Hi @Pegasusx10, Please take a look and make the neccessary changes

**Pull request merged**